### DISCUSSION TITLE
Zero values "omitted" - omitted as in setting them to NAs?

### DISCUSSION THREAD

**Discussion by Mengbo-Li** [vdemichev/DiaNN#530] [Q&A] [answered]
Hi Vadim, 

Thanks for the great package and also thanks for answering questions here, it's been very helpful. I just have a quick question:

In the README, there is a statement of "Sometimes DIA-NN will report a zero as the best estimate for a precursor or protein quantity. Such zero quantities are omitted from protein/gene matrices. "

- For the zero-valued precursors, do you mean they are set as NAs when performing MaxLFQ to get the corresponding columns in the main report?
- For the zero-valued protein/gene-level quantities, are they left as NAs in the output matrices?

Cheers, 
Mengbo

**Comment by vdemichev** [comment, marked as answer]
Hi Mengbo,

Yes, 0 precursor values are ignored when performing MaxLFQ.
Actually, MaxLFQ quantities cannot be 0-value. The regular protein quantities yes - can. And these are omitted from matrices.

Best,
Vadim

**Reply by Mengbo-Li** [reply]
Wow thanks for the quick reply!

**Reply by Mengbo-Li** [reply]
Sorry for the follow up question, this is regarding the matrix output compared with the main report:

I notice that `pg_matrix.tsv` is a submatrix of what I get if I extract the `PG.MAXLFQ` column from the main report. I am just wondering in addition to Q.value <= 0.01 and PG.Q.Value < 0.01, are there any other additional filtering applied to obtain the `PG.MAXLFQ` column by the main program?

I am asking this because when I do `diann::diann_maxlfq(df)`, where df is after filtering for Q.value <= 0.01 and PG.Q.Value < 0.01, I get quite different results.

**Reply by vdemichev** [reply]
Yes, please see the pg_matrix specification in the docs

**Reply by Mengbo-Li** [reply]
Right, well I meant that while the pg_matrix is obtained by applying the filters, so that some protein groups will not be in the resulting matrix. Meanwhile in the main report, even for proteins groups that do not appear in the pg_matrix output, they still have a PG.MAXLFQ value. I am just wondering how MAXLFQ is obtained for PG's that are not in the pg_matrix output?

**Reply by vdemichev** [reply]
General recommendation: if in doubt about the contents of pg_matrix, just use the main report :)
